Release 15.5.2 (#55492)
gravitational/teleport: CHANGELOG.md

# Changelog

> ## 15.5.2 (06/05/25)

Reviewer: ```suggestion
## 15.5.2 (06/05/25)

### Security fixes

This patch includes some security fixes.
These issues are present in previous v15 releases.
Impacted users are recommended to upgrade their auth and proxy servers to the latest version.

#### [High] Short to long term access escalation in Okta integration

* Enterprise fix: Verified required Okta OAuth scopes during plugin creation/update.

In Okta integration configurations with enabled access lists sync, a user with an approved  just-in-time access request to an Okta application could be unintentionally promoted to an access list granting access to the same application. This would result in the access to the Okta app/group persisting after the access request expiration.

This vulnerability affects Okta integration users who have access lists sync enabled. You can check whether you have an Okta integration installed with access lists sync enabled either in the Teleport web UI under Zero Trust Access / Integrations page or by running “tctl get plugins/okta” CLI command and looking at the “spec.settings.okta.sync_settings.sync_access_lists” flag.

#### [Medium] Unsanitized user input affecting PKINIT (Kerberos)

* Fixed users being able to overwrite host files on SQL Server and Oracle
  database systems when using Kerberos. [#55143](https://github.com/gravitational/teleport/pull/55143)

This vulnerability affects users who authenticate to SQL Server and Oracle with
Kerberos using the PKINIT flow.

### Other fixes and improvements
```

Author: We never did a private release for v15. That enterprise fix you have added text for is not in v15 teleport.e. The PKINIT fix was merged May 26, so I guess that was not a private release fix?

> * Prevented unknown resource kinds from rendering errors in the web UI. [#55211](https://github.com/gravitational/teleport/pull/55211)

Reviewer: ```suggestion
* Fixed Unknown resource kinds rendering errors in the web UI. [#55211](https://github.com/gravitational/teleport/pull/55211)
```

> * Fixed formatting of the `cache_component` label for the Prometheus metrics `teleport_cache_health` and `teleport_cache_last_reset_seconds`. [#55193](https://github.com/gravitational/teleport/pull/55193)

Reviewer: ```suggestion
* Fixed the formatting of the `cache_component` label for the Prometheus metrics `teleport_cache_health` and `teleport_cache_last_reset_seconds`. [#55193](https://github.com/gravitational/teleport/pull/55193)
```

> * Fixed the impact of malicious `--db-user` values on PKINIT flow. [#55144](https://github.com/gravitational/teleport/pull/55144)

Reviewer: ```suggestion
```

> * Fixed an issue database discovery fails when there are more than 5 OpenSearch domains. [#55060](https://github.com/gravitational/teleport/pull/55060)

Reviewer: ```suggestion
* Fixed database discovery failing when there are more than 5 OpenSearch domains. [#55060](https://github.com/gravitational/teleport/pull/55060)
```
